internal/provider/catalog/media_datasource.go

```diff
@@ -0,0 +1,215 @@
+				Validators: []validator.String{
+					stringvalidator.ExactlyOneOf(path.MatchRoot("catalog_id")),
+				},
+			},
+			"name": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "Media name.",
+			},
+			"description": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "Media description.",
+			},
+			"is_iso": schema.BoolAttribute{
+				Computed:            true,
+				MarkdownDescription: "True if this media file is an Iso.",
+			},
+			"owner_name": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "Owner name.",
+			},
+			"is_published": schema.BoolAttribute{
+				Computed:            true,
+				MarkdownDescription: "True if this media file is in a published catalog.",
+			},
+			"created_at": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "Creation date.",
+			},
+			"size": schema.Int64Attribute{
+				Computed:            true,
+				MarkdownDescription: "Media size in bytes.",
+			},
+			"status": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "Media status.",
+			},
+			"storage_profile": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "Name of the storage profile.",
+			},
+		},
+	}
+}
+
+func (d *catalogMediaDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*client.CloudAvenue)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *client.CloudAvenue, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	d.client = client
+}
+
+func (d *catalogMediaDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var data catalogMediaDataSourceModel
+	// Read Terraform configuration data into the model
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Check if catalog_id is set
+	if data.CatalogID.IsNull() {
+		// If not, try to find it using catalog_name
+		catalog, err := d.client.Vmware.Client.GetCatalogByName(d.client.Org, data.CatalogName.ValueString())
+		if err != nil {
+			tflog.Error(ctx, "Unable to find catalog.")
+			resp.Diagnostics.AddError("Unable to find catalog", err.Error())
+			return
+		}
+		data.CatalogID = types.StringValue(catalog.Catalog.ID)
+	}
+
+	catalog, err := d.client.Vmware.Client.GetCatalogById(data.CatalogID.ValueString())
+	if err != nil {
+		tflog.Error(ctx, "Unable to find catalog.")
+		resp.Diagnostics.AddError("Unable to find catalog", err.Error())
+		return
+	}
+
+	var media *govcd.Media
+	identifier := data.Name.ValueString()
+	media, err = catalog.GetMediaByNameOrId(identifier, false)
```

Comment:
var `identifier` is useless

```suggestion
	media, err = catalog.GetMediaByNameOrId(data.Name.ValueString(), false)
```